libkernelflinger/log.c

```diff
@@ -140,6 +143,67 @@ static EFI_STATUS serial_init()
 	return EFI_SUCCESS;
 }
 
+EFI_FILE *log_open_output_file(const CHAR16 *filename)
```

Comment:
This function should just have `EFI_HANDLE *handle` parameter in addition to the filename and you should get rid of the `g_root `global variable.  Global variable should be limited to situation where there are no other good alternative (almost never).

Reply:
Already modified base on the suggestion.